File: asymmetric/asymmetric.go
package asymmetric

import (
	"golang.org/x/crypto/curve25519"
	"github.com/mabels/neckless/key"
)

func CreateShared(priv *key.RawKey, pub *key.RawKey) key.RawKey {
	shared := [32]byte{}
	curve25519.ScalarMult(&shared, priv.As32Byte(), pub.As32Byte())
	return shared
}

File: asymmetric/asymmetric.ts
import { RawKey } from '../key/key'

export function CreateShared(priv: RawKey, pub: RawKey): RawKey {
	// curve25519.ScalarMult(&shared, priv.As32Byte(), pub.As32Byte())
	throw Error("CreateShared not implemented")
	const shared = new RawKey(Buffer.from([]))
	return shared
}
